component/src/main/java/org/wso2/extension/siddhi/store/cassandra/CassandraEventTable.java

```
@@ -0,0 +1,1147 @@
+                if (ids.isEmpty()) {
+                    // Since we do not know the primary key this operation cannot be done
+                    throw new CassandraTableException("No results found for the given values. Only update " +
+                            "functionality is capable without primary keys. If update or insert operation is needed" +
+                            "whole key should be included");
+                } else {
+                    executeAsBatchNonPrimaryKeyUpdate(ids, updateSetParameterMap);
+                }
+                i++;
+            }
+
+        }
+    }
+
+    @Override
+    protected CompiledCondition compileCondition(ExpressionBuilder expressionBuilder) {
+        CassandraConditionVisitor visitor = new CassandraConditionVisitor();
+        expressionBuilder.build(visitor);
+        return new CassandraCompiledCondition(visitor.returnCondition(), visitor.getParameters(),
+                visitor.getReadOnlyCondition(), visitor.getStoreAtrributeNames());
+    }
+
+    @Override
+    protected CompiledExpression compileSetAttribute(ExpressionBuilder expressionBuilder) {
+        return compileCondition(expressionBuilder);
+    }
+
+    @Override
+    protected void connect() throws ConnectionUnavailableException {
+        //creating Cluster object
+        String username = storeAnnotation.getElement(ANNOTATION_USER_NAME);
+        String password = storeAnnotation.getElement(ANNOTATION_PASSWORD);
+        Cluster cluster;
+        if (username.isEmpty() && password.isEmpty()) {
+            cluster = Cluster.builder().addContactPoint(host).withPort(port).build();
+        } else {
+            cluster = Cluster.builder().addContactPoint(host).withCredentials(username, password).build();
+        }
+        // NoHostAvailableException: All host(s) tried for query failed   when the connection is not established..
```

Exceptions are not handled?